elfi/visualization/progress_bar.py

```diff
@@ -0,0 +1,55 @@
+"""Progress bar for inference methods."""
+
+from math import ceil
+
+from IPython.display import display
+from ipywidgets import HTML, IntProgress, VBox
+
+
+class ProgressBar(object):
+    """Progress bar for showing the inference process."""
+
+    def __init__(self, batch_size, n_samples, sampler, quantile=None, n_sim=None):
```

Comment:
Making the progress bar depend on all this info doesn't make it very general. 

Testing with a `print` in `ParameterInference.infer` should give you an idea for a rather general solution (inspired by `ParameterInference.finished`):

```
while not self.finished:
    print('Progress: ', self.state['n_batches'], self._objective_n_batches)
    self.iterate()
```

If you always update the ProgressBar with both ` self.state['n_batches']` and  `self._objective_n_batches`, it should work properly in all cases. With threshold-based methods users may sometimes observe "negative" progress, but I guess it's fine.